pkg/controller/storage/capability/capability_controller.go

```diff
@@ -356,7 +364,7 @@ func (c *StorageCapabilityController) deleteStorageCapability(name string) error
 }
 
 func (c *StorageCapabilityController) deleteSnapshotClass(name string) error {
-	if !c.snapshotSupported {
+	if !c.snapshotSupported || c.snapshotClassClient == nil {
```

Comment:
This line can be refine to an common func

Reply:
OK

Reply:
We already have unit test ‘TestCreateStorageClassNotSupportSnapshot ’ for this case.